src/routes/jira/server/jira-server-url-post.test.ts

```diff
@@ -0,0 +1,185 @@
+import { Installation } from "models/installation";
+import express, { Express, NextFunction, Request, Response } from "express";
+import { RootRouter } from "../../router";
+import supertest from "supertest";
+import { getLogger } from "config/logger";
+import { encodeSymmetric } from "atlassian-jwt";
+import { gheServerUrlErrors } from "routes/jira/server/jira-server-url-post";
+import { GitHubServerApp } from "models/github-server-app";
+
+describe("Jira Server Url Suite", () => {
+	let app: Express;
+	let installation: Installation;
+	let jwt: string;
+
+	beforeEach(async () => {
+		installation = await Installation.install({
+			host: jiraHost,
+			sharedSecret: "shared-secret",
+			clientKey: "client-key"
+		});
+		app = express();
+		app.use((req: Request, res: Response, next: NextFunction) => {
+			res.locals = { installation };
+			req.log = getLogger("test");
+			req.session = { jiraHost };
+			next();
+		});
+		app.use(RootRouter);
```

Comment:
Minor suggestions, **can ignore this in this PR**, since we are testing the `jira-server-url-post`, which:
1. Far behind the rootRouter
2. It is not a router anyway, just a function

we can:
1. Don't use `app: express` and `supertest` at all.
2. Ignore preparing `jwt = encodeSymmetric({`
3. And just call that function and test output.

Reply:
Oh, happy to be disagreed on this one though. Would love to trigger some heathy discussion on testing.

Reply:
Because we don't control the input going into the function, we are testing the whole stack that relates to the route and were able to find issues because of this method.  If we didn't, we could have changed the route to be a GET requests instead, or have a type in the path, or that the authentication doesn't work, etc etc

Using supertest doesn't really add any extra overhead, but it does give us absolute confidence that everything is setup correctly in express and the function is triggered as it should be.

Reply:
`If we didn't, we could have changed the route to be a GET requests instead` <-- that's exactly why this test is not complete (but doesn't need to be).

There's a `JiraServerUrlRouter`, in which it calls 
```
JiraServerUrlRouter.route("/").post(JiraContextJwtTokenMiddleware, JiraServerUrlPost);
```

"We could have changed the route to be a GET requests instead" in above, and this test still green, therefore not a valid test.

Reply:
I'm not sure what you're arguing?  If you change that function to be a GET request instead of a POST, it will fail the test...

Reply:
All good then.